File: sum_cube_digits.py
def cube( n ):
	
	s = 0
	while n != 0:
		m = n % 10
		s += m * m * m
		n //= 10
	return s    
	
def main():
	
	n = int(input("Enter a number = "))
	
	arr = [0] * 100
	
	k = 0
	
	arr[ k ] = n
	
	found = False;
	
	while not found:
		
	      k += 1
	      
	      arr[ k ] = cube(arr[k - 1]) 
	      
	      for i in range(0, k):
	      	
	      	  if arr[i] == arr[k]:
	      	  	
	      	  	 found = True
	      	  	 
	for i in range(0, k + 1):
		
		print(arr[i], end = " ")
main()		

File: sum_cube_digits.cpp
#include <iostream>

using namespace std;

typedef unsigned int uint;

uint makeCube(uint n) {
	
	uint cube = 0;
	
	while( n ) {
		
		uint m = n % 10;
		
		cube += m * m * m;
		
		n /= 10;
	}
	
	return cube;
}

int main() {
	
    uint n, k;	
    
	cin>>n;
	
	uint arr[100];
	
	uint found = 0;
	
	k = 0;
	
	arr[ k ] = n;
	 
	do { 
	k++; 
	arr[ k ] = makeCube(arr[ k - 1 ]);
	
	for(int i = 0; i < k; ++i) 
	
	   if(arr[k] == arr[i]) found = 1;
	   
	}while(!found);	
	
	for(int i = 0; i < k; ++i) cout<<arr[i]<<" ";
	   
	cout<<arr[1];
	
	return 0;
}
